test/csv/parse/test_general.rb

```diff
@@ -138,28 +138,71 @@ def test_non_regex_edge_cases
     end
   end
 
-  def test_malformed_csv_cr_first_line
-    error = assert_raise(CSV::MalformedCSVError) do
-      CSV.parse_line("1,2\r,3", row_sep: "\n")
+  def test_unquoted_cr_with_lf_row_separator
+    data = "field1,field\rwith\rcr,field3\nrow2,data,here\n"
+    expected = [
+      ["field1", "field\rwith\rcr", "field3"],
+      ["row2", "data", "here"]
+    ]
+    assert_equal(expected, CSV.parse(data, row_sep: "\n"))
+  end
+
+  def test_unquoted_cr_with_custom_row_separator
+    data = "field1,field\rwith\rcr,field3|row2,data,here|"
+    expected = [
+      ["field1", "field\rwith\rcr", "field3"],
+      ["row2", "data", "here"]
+    ]
+    assert_equal(expected, CSV.parse(data, row_sep: "|"))
+  end
+
+  def test_unquoted_cr_with_crlf_row_separator
+    data = "field1\r,field2,field3\r\nrow2,data,here\r\n"
+    assert_raise(CSV::MalformedCSVError) do
+      CSV.parse(data, row_sep: "\r\n")
     end
-    assert_equal("Unquoted fields do not allow new line <\"\\r\"> in line 1.",
-                 error.message)
   end
 
-  def test_malformed_csv_cr_middle_line
-    csv = <<-CSV
-line,1,abc
-line,2,"def\nghi"
+  def test_unquoted_cr_rejected_when_included_in_row_separator
+    data = "field1,field\r2,field3\r\nrow2,data,here\r\n"
+    assert_raise(CSV::MalformedCSVError) do
+      CSV.parse(data, row_sep: "\r\n")
+    end
+  end
```

Comment:
Are they the same concept tests?
Can we remove `test_unquoted_cr_rejected_when_included_in_row_separator`?